Verify SSE settings checksum before and after uploading an object (#745)
awslabs/mountpoint-s3: doc/CONFIGURATION.md

### Data encryption

Amazon S3 supports a number of [server-side encryption types](https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingEncryption.html). Mountpoint supports reading and writing to buckets that are configured with Amazon S3 managed keys (SSE-S3), with AWS KMS keys (SSE-KMS), or with dual-layer encryption with AWS KMS keys (DSSE-KMS) as the default encryption method. It does not currently support reading objects encrypted with customer-provided keys (SSE-C).

> New objects can be encrypted using different SSE settings than the bucket's default. To use a different setting, the argument `--sse <aws:kms|aws:kms:dsse>` should be used to specify the SSE encryption type, and `--sse-kms-key-id <KEY_ARN>` may be used to optionally specify a KMS key ID. When the `--sse` argument is used without providing a KMS key ID, S3 will use an [AWS managed KMS key](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#key-mgmt), which is created automatically. Please note that these command line arguments only configure server-side encryption for *new* objects created with Mountpoint, all of the *existing* objects will remain encrypted with settings specified at the time of their creation.

Reviewer: Yep, that's important. Let's move these docs to a follow-up PR.

> New objects can be encrypted using different SSE settings than the bucket's default. To use a different setting, the argument `--sse <aws:kms|aws:kms:dsse>` should be used to specify the SSE encryption type, and `--sse-kms-key-id <KEY_ARN>` may be used to optionally specify a KMS key ID. When the `--sse` argument is used without providing a KMS key ID, S3 will use an [AWS managed KMS key](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#key-mgmt), which is created automatically. Please note that these command line arguments only configure server-side encryption for *new* objects created with Mountpoint, all of the *existing* objects will remain encrypted with settings specified at the time of their creation.

Author: makes sense, good that we crisped that already though

> If you would like to encrypt new objects written with Mountpoint using a different server-side encryption settings than the bucket's default, you can use `--sse <aws:kms|aws:kms:dsse>` and `--sse-kms-key-id example-key-id` command-line arguments. You can only override bucket's default SSE type with either `aws:kms` or `aws:kms:dsse`. When you are using one of those types, server-side encryption may be further configured to use a specific KMS key ID with a `--sse-kms-key-id` argument. When an `--sse` argument is used without a KMS key ID, S3 will use an [AWS managed KMS key](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#key-mgmt), which is created automatically. Please note, that these command line arguments do only configure server-side encryption for *new* objects, all of the *existing* objects will remain encrypted with settings specified on their creation.

Reviewer: A few small nits. We can also use more passive voice here.

```suggestion
New objects can be encrypted using different SSE settings than the bucket's default. To use a different setting, the argument `--sse <aws:kms|aws:kms:dsse>` should be used to specify the SSE encryption type, and `--sse-kms-key-id <KEY_ARN>` may be used to optionally specify a KMS key ID. When the `--sse` argument is used without providing a KMS key ID, S3 will use an [AWS managed KMS key](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#key-mgmt), which is created automatically. Please note that these command line arguments only configure server-side encryption for *new* objects created with Mountpoint, all of the *existing* objects will remain encrypted with settings specified at the time of their creation.
```

Author: fixed
